File: backend/app/schemas/career_ready.py
import datetime as dt
from typing import Literal, Optional

from pydantic import BaseModel, Field


class CareerReadyAnalysisItem(BaseModel):
    session_id: str
    discipline: str
    completed_at: dt.datetime

    public_summary: str = Field(min_length=1, max_length=500)

    conceptual_consistency_score: int
    technical_adequacy_score: int
    originality_score: int

    avg_score: float


class CareerReadyMineResponse(BaseModel):
    display_name: str
    required_reviews: int
    completed_reviews: int
    progress_percent: int

    target_label: Literal["Kariyere Hazırlık Analizi"] = "Kariyere Hazırlık Analizi"

    avg_score: Optional[float] = None
    items: list[CareerReadyAnalysisItem]


class CareerReadyShareResponse(BaseModel):
    share_display_name: str
    required_reviews: int
    completed_reviews: int
    progress_percent: int
    target_label: Literal["Kariyere Hazırlık Analizi"] = "Kariyere Hazırlık Analizi"

    avg_score: Optional[float] = None
    items: list[CareerReadyAnalysisItem]


class CareerReadyShareTokenResponse(BaseModel):
    token: str

File: shared/contracts/career_ready.ts
export type CareerReadyAnalysisItem = {
  session_id: string
  discipline: string
  completed_at: string

  public_summary: string

  conceptual_consistency_score: number
  technical_adequacy_score: number
  originality_score: number
  avg_score: number
}

export type CareerReadyMineResponse = {
  display_name: string
  required_reviews: number
  completed_reviews: number
  progress_percent: number
  target_label: "Kariyere Hazırlık Analizi"

  avg_score: number | null
  items: CareerReadyAnalysisItem[]
}

export type CareerReadyShareTokenResponse = {
  token: string
}

export type CareerReadyShareResponse = {
  share_display_name: string
  required_reviews: number
  completed_reviews: number
  progress_percent: number
  target_label: "Kariyere Hazırlık Analizi"

  avg_score: number | null
  items: CareerReadyAnalysisItem[]
}

